### DISCUSSION TITLE
Sakhi-API-Service response history not appended in the context.

### DISCUSSION THREAD

**Discussion by abhilashkadu** [DJP-Digital-Jaaduii-Pitara/sakhi-api-service#126] [Q&A] [answered]
When the query API from the Sakhi-API-service is called, a new context is created each time. The context does not retain the previous response history and loses the context, which results in the answer quality not improving. Are there any settings available to enable history retention?

**Comment by Jayaprakash8887** [comment, marked as answer]
Please use `'/v1/chat'` api for conversation kind of workflow. `'/v1/query'` works in Q&A mode only

https://ai-assistant.sunbird.org/get-started-with-ai-assistant/apis
